presto-spark-base/src/main/java/com/facebook/presto/spark/execution/task/PrestoSparkNativeTaskExecutorFactory.java

```diff
@@ -503,7 +506,9 @@ public Tuple2<MutablePartitionId, T> next()
             checkArgument(outputType == PrestoSparkSerializedPage.class,
                     format("PrestoSparkNativeTaskExecutorFactory only outputType=PrestoSparkSerializedPage" +
                             "But tried to extract outputType=%s", outputType));
-            return new Tuple2<>(new MutablePartitionId(), (T) toPrestoSparkSerializedPage(next.get()));
+            MutablePartitionId mutablePartitionId = new MutablePartitionId();
+            mutablePartitionId.setPartition(partitionId);
```

Comment:
can you explain this change? don't have context here, but seems we're returning a different value.

Reply:
Currently this value is not used. I'm setting to partition ID to allow for matching results to the task on the driver. This allow the logging above to include task ID:

```
log.info("Received %s rows from taskId %s in fragment %s", page.getPositionCount(), tuple._1.getPartition(), inputFuture.getKey());
```

Otherwise 'tuple._1.getPartition()' would be empty.